src/common/libsubprocess/test/iostress.c

```diff
@@ -0,0 +1,310 @@
+        }
+        flux_watcher_stop (w);
+    }
+    return;
+error:
+    free (buf);
+    //flux_reactor_stop_error (r);
+    iostress_start_doomsday (ctx, 2.);
+}
+
+flux_subprocess_ops_t iostress_ops = {
+    .on_completion      = iostress_completion_cb,
+    .on_state_change    = iostress_state_cb,
+    .on_stdout          = iostress_output_cb,
+};
+
+bool iostress_run_check (flux_t *h,
+                         const char *name,
+                         bool direct,
+                         int stdin_bufsize,
+                         int stdout_bufsize,
+                         int batchcount,
+                         int batchlines,
+                         size_t linesize)
+{
+    char *cat_av[] = { "cat", NULL };
+    flux_cmd_t *cmd;
+    struct iostress_ctx ctx;
+    int rc;
+    bool ret = true;
+    char nbuf[16];
+
+    memset (&ctx, 0, sizeof (ctx));
+    ctx.h = h;
+    ctx.batchcount = batchcount;
+    ctx.batchlines = batchlines;
+    ctx.linesize = linesize;
+    ctx.name = name;
+    ctx.direct = direct;
+
+    if (!(cmd = flux_cmd_create (ARRAY_SIZE (cat_av) - 1, cat_av, environ)))
+        BAIL_OUT ("flux_cmd_create failed");
+    if (stdin_bufsize > 0) {
+        snprintf (nbuf, sizeof (nbuf), "%d", stdin_bufsize);
+        if (flux_cmd_setopt (cmd, "stdin_BUFSIZE", nbuf) < 0)
+            BAIL_OUT ("flux_cmd_setopt failed");
+    }
+    if (stdout_bufsize > 0) {
+        snprintf (nbuf, sizeof (nbuf), "%d", stdout_bufsize);
+        if (flux_cmd_setopt (cmd, "stdout_BUFSIZE", nbuf) < 0)
+            BAIL_OUT ("flux_cmd_setopt failed");
+    }
+    if (!(ctx.p = flux_rexec (h, FLUX_NODEID_ANY, 0, cmd, &iostress_ops)))
+        BAIL_OUT ("flux_rexec failed");
+    if (flux_subprocess_aux_set (ctx.p, "ctx", &ctx, NULL) < 0)
+        BAIL_OUT ("flux_subprocess_aux_set failed");
+
+    if (!(ctx.source = flux_prepare_watcher_create (flux_get_reactor (h),
+                                                    iostress_source_cb,
+                                                    &ctx)))
+        BAIL_OUT ("could not create prepare watcher");
+    if (!(ctx.timer = flux_timer_watcher_create (flux_get_reactor (h),
+                                                 0.,
+                                                 0.,
+                                                 iostress_timer_cb,
+                                                 &ctx)))
+        BAIL_OUT ("counld not create timer watcher");
+
+    rc = flux_reactor_run (flux_get_reactor (h), 0);
+    if (rc < 0) {
+        diag ("%s: flux_reactor_run: %s", name, strerror (errno));
+        ret = false;
+    }
+
+    diag ("%s: processed %d of %d lines", name,
+          ctx.linerecv, ctx.batchcount * ctx.batchlines);
+    if (ctx.linerecv < ctx.batchcount * ctx.batchlines)
+        ret = false;
+
+    flux_watcher_destroy (ctx.source);
+    diag ("%s: destroying subprocess", name);
+    flux_subprocess_destroy (ctx.p);
+    flux_cmd_destroy (cmd);
+
```

Comment:
Minor: is `ctx.timer` leaked here?

Reply:
Good catch!  Still playing with the tests a bit.  They are a little tricky to get right.

Reply:
Sorry! I neglected to notice this PR was still WIP.

Reply:
No problem!  I should be wrapping up as soon as I get a clean CI run.